File: codeforces/div-3-1980/C.py
t = int(input())
for _ in range(0, t):
  n = int(input())
  a = [int(_) for _ in input().split()]
  b = [int(_) for _ in input().split()]
  m = int(input())
  d = [int(_) for _ in input().split()]

  if d[m-1] not in b:
    print("No")
    continue

  c = []
  indx = 0
  while indx<n: 
    if a[indx]!=b[indx]:
      c.append(b[indx])
    indx += 1
  
  c = sorted(c)
  d = sorted(d)

  ic = 0
  id = 0
  while ic<len(c) and id<len(d):
    if d[id]==c[ic]:
      ic+=1
      id+=1
    elif d[id]<c[ic]:
      id+=1
    else:
      break

  if ic==len(c):
    print("Yes")
  else:
    print("No")

File: codeforces/div-3-1980/C.rb
gets.to_i.times do 
  n = gets.to_i 
  a = gets.strip.split(' ').map(&:to_i)
  b = gets.strip.split(' ').map(&:to_i)
  m = gets.to_i
  d = gets.strip.split(' ').map(&:to_i)
  if not b.include?(d.last) then 
    puts("No")
    next
  end

  i = 0
  c = []
  while i<n do
    c<<b[i] if a[i]!=b[i] 
    i+=1
  end

  c = c.sort 
  d = d.sort 

  id = 0
  ic = 0
  while id<d.size and ic<c.size do 
    if d[id]==c[ic] then 
      id+=1
      ic+=1
    elsif d[id]<c[ic] then 
      id += 1
    else
      break
    end
  end

  if ic==c.size then 
    puts("Yes")
  else
    puts("No")
  end
end
